profiler/profile.go

```diff
@@ -175,6 +181,28 @@ var profileTypes = map[ProfileType]profileType{
 			return buf.Bytes(), err
 		},
 	},
+	executionTrace: {
+		Name:     "execution-trace",
+		Filename: "trace.gotrace", // TODO: pick a good name
+		Collect: func(p *profiler) ([]byte, error) {
+			if !p.shouldTrace() {
+				return nil, errors.New("execution trace collection is disabled")
+			}
+			defer func() {
+				// TODO: this updates the last trace time
+				// regardless of whether tracing suceeds or not.
+				// Should we only update it if tracing succeeds?
+				p.lastTrace = time.Now()
```

Comment:
If we _don't_ update the last trace time when tracing fails, we'll try again in the next profiling period (i.e. in 1 minute) rather than 5000 seconds later. So not updating the time actually would make us attempt tracing more often. Given that, maybe uncondionally updating the time is better if overhead is the concern?

Like you said, the error only happens if tracing was enabled somewhere else already, like with CPU profiling. I'd consider that kind of error to be a sign of misuse. Tracing and profiling are global things, and if a user goes through this library then they shouldn't enable tracing/profiling elsewhere.

Reply:
Oh, I'm sorry. I got myself confused. Yes, the current approach is the right one. We want to guarantee that we're no attempting to start the trace more often than promised!